# T2: Adult Data Analysis

In [1]:
import pandas as pd
import numpy as np
import os

In [2]:
#adult = pd.read_excel('S://TANF Reports//Reports Analytics//202112//202112-t_worc_adults-20220215_1356.xlsx')
adult = pd.read_excel('S://TANF Reports//!TANF Federal Reports//202201//WORC//Version4//202201-t_worc_adults-20220504_0859.xlsx')

In [3]:
adult['dob'] = pd.to_datetime(adult['dob'].astype('str'), infer_datetime_format = True)
adult['reporting_month'] = pd.to_datetime(adult['reporting_month'].apply(lambda x: f"{x}01").astype(str), infer_datetime_format=True)
adult['age'] = np.floor((adult['reporting_month'] - adult['dob'])/np.timedelta64(1,'Y'))

In [4]:
adult.shape

(1253, 74)

In [5]:
ncp = adult['ncp'].value_counts()
ncp

Not a noncustodial parent    1253
Name: ncp, dtype: int64

In [6]:
ncp_list = adult.loc[adult['ncp'] == "Yes, a noncustodial parent",('case_number','ncp')]
df_ncp_list = pd.DataFrame(ncp_list)
df_ncp_list

Empty DataFrame
Columns: [case_number, ncp]
Index: []

In [7]:
adult['work_partic_status'].value_counts()

Not applicable (e.g., not a work-eligible individual; person living in household and whose income or resources are counted in determining eligibility for or the amount of assistance for the family receiving assistance, but is not in the eligible family receiving assistance or noncustodial parent that the State opted to exclude in determining participation rate).                                                                                                                                                                                                                                                                  772
 Required to participate, but not participating; and not subject to a work sanction for the reporting month.                                                                                                                                                                                                                                                                          

In [8]:
adult['age'].describe()

count    1253.000000
mean       41.162011
std        13.507426
min        18.000000
25%        30.000000
50%        39.000000
75%        52.000000
max        84.000000
Name: age, dtype: float64

In [9]:
adult['marital_status'].value_counts()

Single, never married       778
Married, living together    237
Married, but separated      110
Divorced                    108
Widowed                      20
Name: marital_status, dtype: int64

In [10]:
adult['provid_child_care'].value_counts()

0    1253
Name: provid_child_care, dtype: int64

In [11]:
adult['parent_with_minor'].value_counts()

Not Parent with minor child in the family                                                        772
Yes, a parent with a minor child in the family, but not used in two-parent participation rate    423
Yes, a parent with a minor child in the family and used in two-parent participation rate          58
Name: parent_with_minor, dtype: int64

4.5% (108)of total cases (2398) are 2 Parent cases <br>
    

95.5% (2290)of total cases are single parent cases